packages/aws-client/src/StepFunctions.ts

```diff
@@ -84,46 +95,48 @@ export const executionExists = (executionArn: string) =>
  * If a ThrottlingException is received, this function will retry using an
  * exponential backoff.
  *
- * @param {Object} params
- * @returns {Promise<Object>}
+ * @param {GetExecutionHistoryInput} params
+ * @returns {Promise<GetExecutionHistoryOutput>}
  *
  * @kind function
  */
-export const getExecutionHistory = improveStackTrace(
-  retryOnThrottlingException(
-    async (
-      params: AWS.StepFunctions.GetExecutionHistoryInput,
-      previousResponse: { events: AWS.StepFunctions.HistoryEventList } = {
-        events: [],
-      }
-    ): Promise<{ events: AWS.StepFunctions.HistoryEventList }> => {
-      const response = await sfn().getExecutionHistory(params).promise();
-      const events = [
-        ...previousResponse.events,
-        ...response.events,
-      ];
-      // If there is a nextToken, recursively call this function to get all events
-      // in the execution history.
-      if (response.nextToken) {
-        return getExecutionHistory({
-          ...params,
-          nextToken: response.nextToken,
-        }, {
-          events,
-        });
-      }
-      return {
+export const getExecutionHistory = retryOnThrottlingException(
+  async (
+    params: GetExecutionHistoryInput,
+    previousResponse: { events: HistoryEvent[] } = {
+      events: [],
+    }
+  ): Promise<{ events: HistoryEvent[] }> => {
+    const response: GetExecutionHistoryOutput
+      = await sfn().getExecutionHistory(params);
+
+    response.events = response.events || [];
+    const events = [
+      ...previousResponse.events,
+      ...response.events,
+    ];
+    // If there is a nextToken, recursively call this function to get all events
+    // in the execution history.
+    if (response.nextToken) {
+      return getExecutionHistory({
+        ...params,
+        nextToken: response.nextToken,
+      }, {
         events,
-      };
+      });
     }
-  )
+    return {
+      events,
+    };
+  }
 );
 
 export const getExecutionStatus = async (executionArn: string) => {
-  const [execution, executionHistory] = await Promise.all([
-    describeExecution({ executionArn }),
-    getExecutionHistory({ executionArn }),
-  ]);
+  const [execution, executionHistory]
+    = await Promise.all([
+      describeExecution({ executionArn }),
+      getExecutionHistory({ executionArn }),
+    ]);
```

Comment:
Do we have some auto-formatting happening here? Might be worth a quick convo.

Reply:
i'll connect with you on slack